packages/docmanager-extension/src/index.ts

```diff
@@ -520,36 +520,21 @@ function addCommands(
     label: () => 'Save All',
     caption: 'Save all open documents',
     isEnabled: () => {
-      const iterator = shell.widgets('main');
-      let widget = iterator.next();
-      while (widget) {
-        let context = docManager.contextForWidget(widget);
-        if (
-          context &&
-          context.contentsModel &&
-          context.contentsModel.writable
-        ) {
-          return true;
-        }
-        widget = iterator.next();
-      }
-      // disable saveAll if all of the widgets models
-      // have writable === false
-      return false;
+      return some(
+        map(shell.widgets('main'), w => docManager.contextForWidget(w)),
+        c => c && c.contentsModel && c.contentsModel.writable
+      );
```

Comment:
🤓 
This is expressive and concise.

Reply:
higher-order iteration functions for the win! :)